Feat docker compose

jotabulacios opened the pull request:
This PR establishes a dockerized environment 
It incorporates Dockerfiles for the following components:
- Opetator 
- Aggretor
- Anvil tesnet

## Type of change

- [x] New 


To test it, run:

```bash
docker-compose build
```

then

```bash
docker-compose up
```

Commit:
add dockerfile for operator

Commit:
add dockerfile for devnet

Commit:
add dockerfile for aggregator

jotabulacios committed:
Update Dockerfile

jotabulacios committed:
add dockerfile for anvil

isoprophlex committed:
Modify aggregator Dockerfile to support foundry

isoprophlex committed:
Fix anvil/state Dockerfile

isoprophlex committed:
Add docker-compose

isoprophlex committed:
Merge branch 'main' into feat-docker-compose

isoprophlex committed:
Merge branch 'origin/main' into docker-compose

isoprophlex committed:
Update aggregator/Dockefile

isoprophlex committed:
Fix operator/config.yaml

isoprophlex committed:
Update work in progress

isoprophlex committed:
Fix docker-compose.yml

Fixed docker-compose.yml and now every container is accessible from another container

isoprophlex committed:
Add two instances of operator in docker-compose

MauroToscano committed:
Merge branch 'main' into feat-docker-compose

taturosati commented on contracts/scripts/anvil/state/Dockerfile:
```diff
@@ -0,0 +1,26 @@
+# Update packages and install required dependencies
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y python3 python3-pip git nano curl iputils-ping redis-tools autossh nmap apt-utils supervisor
```
Why installing so many dependencies?

isoprophlex replied:
I deleted a few a seconds ago, but python, git and curl are necessary to install foundry or eigenlayer.

taturosati commented on aggregator/Dockerfile:
```diff
@@ -0,0 +1,36 @@
+# Install system dependencies
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get install -y python3 python3-pip git nano curl iputils-ping redis-tools autossh nmap apt-utils supervisor
```
Why installing so many dependencies?

isoprophlex replied:
I deleted a few a seconds ago, but python, git and curl are necessary to install foundry or eigenlayer.

taturosati commented on operator/config.yaml:
```diff
@@ -0,0 +1,12 @@
+private_key_store_path: "/Users/franco/.eigenlayer/operator_keys/operator.ecdsa.key.json"
+bls_private_key_store_path: "/Users/franco/.eigenlayer/operator_keys/operator.bls.key.json"
+signer_type: local_keystore
+chain_id: 31337
```
Add missing new line

taturosati commented on operator/Dockerfile:
```diff
@@ -0,0 +1,15 @@
+# Copy the binary from the builder stage to the current stage.
+COPY --from=builder /app/operator-app /operator-app
+# Set the binary as the entrypoint of the container.
+ENTRYPOINT ["/operator-app"]
```
Need to update to pass in config-file flag

taturosati replied:
+ Missing new line at end of file

JuArce commented on docker-compose.yml:
```diff
@@ -0,0 +1,40 @@
+      dockerfile: operator/Dockerfile
+    depends_on:
+      - devnet
+    network_mode: host
```
missing new line

isoprophlex committed:
Merge 'origin/main' into feat-docker-compose

isoprophlex committed:
Add whitespaces in required files

jotabulacios committed:
add operator to the dockerfile

isoprophlex committed:
Delete two operators

I deleted two operators in docker-compose to make easier testing if operator works.

isoprophlex committed:
Add first approach to operator Dockerfile

isoprophlex committed:
Add full-registration for operators

MauroToscano commented:
Stale
